src/libraries/System.Security.Cryptography.Csp/tests/RSACryptoServiceProviderProvider.cs

```diff
@@ -18,6 +21,8 @@ public class RSACryptoServiceProviderProvider : IRSAProvider
         public bool SupportsSha2Oaep => false;
 
         public bool SupportsPss => false;
+
+        public bool SupportsSha1Signatures => _supportsSha1Signatures ??= SignatureSupport.CanProduceSha1Signature(Create());
```

Comment:
In RSACryptoServiceProviderTests this gets repeated, but using `{ get; } = ...`.  Do we really need it in both places?

Also, is there a reason we need to use explicit memo-ization here, vs just using the property assignment syntax?  (I'll believe "yes, the cctor here runs too early", so we need to delay it, but have to ask)

Reply:
> Do we really need it in both places?

🤔 no.

> Also, is there a reason we need to use explicit memo-ization here

This does not work:

```csharp
public bool SupportsSha1Signatures { get; } = SignatureSupport.CanProduceSha1Signature(Create());
```

> A field initializer cannot reference the non-static field, method, or property 'RSACryptoServiceProviderProvider.Create()' 

and this:

```csharp
public bool SupportsSha1Signatures => SignatureSupport...
```

Would mean we don't memoize it at all, so I went with what worked.